02-microarray/pathway-analysis_microarray_02_ora_with_webgestaltr.Rmd

```diff
@@ -0,0 +1,262 @@
+We are using `head()` so we just get a preview and not the whole table printed out. 
+
+```{r}
+head(listGeneSet("hsapiens"))
+```
+
+The `WebGestaltR()` function will generate an HTML file with a report (path specified by the `outputDirectory` and `projectName` arguments) _and_ return a data frame.
+
+```{r}
+go_enrichment_results <-
+  WebGestaltR(
+    enrichMethod = "ORA",
+    organism = "hsapiens",
+    enrichDatabase = "geneontology_Biological_Process", # What categories we'd like enrichment results for
+    interestGene = upregulated_genes, # The vector of our genes of interest
+    interestGeneType = "ensembl_gene_id",
+    referenceGene = all_genes, # The vector of all the genes originally in our set
+    referenceGeneType = "ensembl_gene_id",
+    outputDirectory = results_dir, # Replace with the directory you would like your results saved to
+    projectName = "GSE37418_up_SHH_lfc2_fdr0_05" # Replace with a relevant output name for the HTML file
+  )
+```
+
+As noted in the messages from `WebGestaltR`, the results are in `results/Project_GSE37418_up_SHH_lfc2_fdr0_05`.
+The HTML report is at [`results/Project_GSE37418_up_SHH_lfc2_fdr0_05/Report_GSE37418_up_SHH_lfc2_fdr0_05.html`](./results/Project_GSE37418_up_SHH_lfc2_fdr0_05/Report_GSE37418_up_SHH_lfc2_fdr0_05.html).
+
+## Read in a results table
+
+Let's take a look at one of the results tables look like by reading it into R. 
+It is a `.txt` file, however its values are separated by tabs, making it a TSV; hence we can read it in with `readr::read_tsv()` function. 
+
+```{r}
+results_df <- readr::read_tsv(
+  file.path(
+    results_dir,
+    "Project_GSE37418_up_SHH_lfc2_fdr0_05",
+    "enrichment_results_GSE37418_up_SHH_lfc2_fdr0_05.txt"
+  )
+)
+```
+
+Let's print out a preview of it.   
+
+```{r}
+head(results_df)
+```
+It looks like there are a lot of pathways associated with ribosomes and translation.
+Gene sets, particularly GO gene sets that are of a hierarchical nature, are not independent, so it's important to keep in mind that many of these gene sets could be telling us the same thing.
+
+In the `link` column, you can use the links to see more information about a particular GO term.
+Take a look at the link for the top result, `translational initiation`: http://amigo.geneontology.org/amigo/term/GO:0006413
+
+# Resources for further learning
+
+- [WebGestalt 2019 Manual](http://www.webgestalt.org/WebGestalt_2019_Manual.pdf)
+- [`WebGestaltR` package documentation](https://www.rdocumentation.org/packages/WebGestaltR/versions/0.3.0).
+- [This handy review](https://doi.org/10.1371/journal.pcbi.1002375) which summarizes the different types of pathway analysis and their limitations [@Khatri2012].
+
+# Session info
+
+At the end of every analysis, before saving your notebook, we recommend printing out your session info. 
+This helps make your code more reproducible by recording what versions of softwares and packages you used to run this. 
```

Comment:
```suggestion
This helps make your code more reproducible by recording what versions of software and packages you used to run this. 
```

Reply:
Yeah. I'll change it here but #244 womp womp.